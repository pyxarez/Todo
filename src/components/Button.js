import React from 'react';

import './Button.res/style.css'

Button.propTypes = {
  type: React.PropTypes.string,
  onClick: React.PropTypes.func,
}

export default function Button({type, onClick, value}) {
  return type
    ? <button
        className={`my-button-component button-${type}`}
        onClick={onClick}>{value}</button>
    : <button
        className={'my-button-component'}
        onClick={onClick}>{value}</button>
}
