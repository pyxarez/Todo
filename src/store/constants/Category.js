export const GET_CATEGORIES = 'GET_CATEGORIES';
export const ADD_CATEGORY = 'ADD_CATEGORY';
export const ADD_NESTED_CATEGORY = 'ADD_NESTED_CATEGORY';
export const DELETE_CATEGORY = 'DELETE_CATEGORY';
export const RENAME_CATEGORY = 'RENAME_CATEGORY';

export const CHANGE_TASK_LOCATION = 'CHANGE_TASK_LOCATION';