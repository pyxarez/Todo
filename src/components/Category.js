import React, { PropTypes } from 'react';
import { Link } from 'react-router';

import Button from './Button';
import './Category.res/style.css';

Category.propTypes = {
  id: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  onExtend: PropTypes.oneOfType(
    [
      PropTypes.func,
      PropTypes.bool
    ]).isRequired,
  children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.element.isRequired),
      PropTypes.bool,
      PropTypes.object
  ])
}

export default function Category({ id, title, children, onExtend }) {
  return (
    <div className="my-category-component">
      {onExtend &&
        <Button type="extend" onClick={onExtend}/>}
      <Link to={`/main/${id}/${title}/`}>{title}</Link>
      {children}
    </div>
  );
}
