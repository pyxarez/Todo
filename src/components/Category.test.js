import React from 'react';
import { shallow } from 'enzyme';

import { Link } from 'react-router';
import Category from './Category';
import Button from './Button';

const setup = () => {
  const props = {
    id: 0,
    title: 'Films',
    onExtend: () => {}
  };

  const wrapper = shallow(
    <Category {...props}>
      <div className='test-class'/>
    </Category>
  );

  return {
    props,
    wrapper
  }
};

describe('Components :: Category', () => {
  it('should render self and subcomponents', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render children', () => {
    const { props } = setup();
    const wrapper = shallow(<Category {...props}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should not render Button component', () => {
    const { props } = setup();
    const newProps = {
      ...props,
      onExtend: false
    };
    const wrapper = shallow(
      <Category {...newProps}>
        <div className="test-class"></div>
      </Category>
     );
    
    expect(wrapper).toMatchSnapshot();
  });
});