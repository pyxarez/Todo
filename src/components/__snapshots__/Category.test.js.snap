exports[`Components :: Category should not render Button component 1`] = `
<div
  className="my-category-component">
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/main/0/Films/">
    Films
  </Link>
  <div
    className="test-class" />
</div>
`;

exports[`Components :: Category should not render children 1`] = `
<div
  className="my-category-component">
  <Button
    onClick={[Function]}
    type="extend" />
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/main/0/Films/">
    Films
  </Link>
</div>
`;

exports[`Components :: Category should render self and subcomponents 1`] = `
<div
  className="my-category-component">
  <Button
    onClick={[Function]}
    type="extend" />
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/main/0/Films/">
    Films
  </Link>
  <div
    className="test-class" />
</div>
`;
